src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.cs

```diff
@@ -162,12 +161,46 @@ public Assembly LoadFromPath(string originalAnalyzerPath)
                 assemblyName = null;
             }
 
+            lock (_guard)
+            {
+                _analyzerAssemblyInfoMap[originalAnalyzerPath] = (assemblyName, realPath, SatelliteCultureNames: null);
+            }
+
+            return (assemblyName, realPath);
+        }
+
+        protected (string RealAssemblyPath, ImmutableHashSet<string> SatelliteCultureNames) GetSatelliteAssemblyInfoForPath(string originalAnalyzerPath)
+        {
+            lock (_guard)
+            {
+                if (!_analyzerAssemblyInfoMap.TryGetValue(originalAnalyzerPath, out var tuple))
+                {
+                    throw new InvalidOperationException();
+                }
+
+                if (tuple is not { } info)
+                {
+                    // GetAssemblyInfoForPath should be called before requesting satellite assembly information
+                    throw new InvalidOperationException();
+                }
```

Comment:
Could just simplify to 
```suggestion
                var tuple = _analyzerAssemblyInfoMap[originalAnalyzerPath];
                if (tuple is not { } info)
                {
                    // GetAssemblyInfoForPath should be called before requesting satellite assembly information
                    throw new InvalidOperationException();
                }
```